ansible-ipi-install/roles/installer/tasks/59_cleanup_bootstrap.yml

```diff
@@ -0,0 +1,43 @@
+---
+- name: Get list of all VMs
+  virt:
+    command: list_vms
+  register: all_vms
+  become: yes
+
+- name: Get list of all running VMs
+  virt:
+    command: list_vms
+    state: running
+  register: running_vms
+  become: yes
+
+- name: Destroy old bootstrap VMs, if any
+  virt:
+    name: "{{ item }}"
+    command: destroy
+  with_items: "{{ running_vms.list_vms }}"
+  when: item is search("bootstrap")
+  become: yes
+
+- name: Undefine old bootstrap VMs, if any
+  virt:
+    name: "{{ item }}"
+    command: undefine
+  with_items: "{{ all_vms.list_vms }}"
+  when: item is search("bootstrap")
+  become: yes
+
+- name: Find old bootstrap VM Storage
+  find:
+    paths: "{{ default_libvirt_pool_dir }}"
+    patterns: '*-bootstrap,*-bootstrap.ign'
+  register: find_results
+  become: yes
+
+- name: Delete old bootstrap VMs Storage
+  file:
```

Comment:
I'm not sure if removing the storage is enough or if `vol-delete` needs to be used. Otherwise I think you can delete them then refreshing the pool.

Reply:
So the reason we're doing it this way is to use an idempotent module rather than `cmd/shell` call to `vol-delete`.  Deleting the files from the disk does in fact leave old data in `virsh vol-list`, temporarily, but this is refreshed during `60_deploy_ocp.yml` when the new bootstrap volumes are created.